File: src/main/java/com/matchFit/post/dto/UpdatePostResponseDto.java
package com.matchFit.post.dto;

import com.matchFit.post.entity.Post;
import com.matchFit.post.entity.Sports;
import com.matchFit.post.entity.Status;
import com.matchFit.post.entity.Town;
import com.matchFit.user.entity.Gender;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class UpdatePostResponseDto {
    private final String title;
    private final String description;
    private final String location;
    private final LocalDateTime date;
    private final int maxPeople;
    private final Gender gender;
    private final Status status;
    private final int cost;
    private final String imageUrl;
    private final Sports sports;
    private final Town town;
    private final LocalDateTime createdAt;
    private final LocalDateTime updatedAt;

    public static UpdatePostResponseDto from(Post post) {
        return new UpdatePostResponseDto(
                post.getTitle(),
                post.getDescription(),
                post.getLocation(),
                post.getDate(),
                post.getMaxPeople(),
                post.getGender(),
                post.getStatus(),
                post.getCost(),
                post.getImageUrl(),
                post.getSports(),
                post.getTown(),
                post.getCreatedAt(),
                post.getUpdatedAt()
        );
    }
}

File: src/main/kotlin/com/matchFit/post/dto/UpdatePostResponseDto.kt
package com.matchFit.post.dto

import com.matchFit.post.entity.Post
import com.matchFit.post.entity.Sports
import com.matchFit.post.entity.Status
import com.matchFit.post.entity.Town
import com.matchFit.user.entity.Gender
import java.time.LocalDateTime

data class UpdatePostResponseDto(
    val title: String,
    val description: String,
    val location: String,
    val date: LocalDateTime,
    val maxPeople: Int,
    val gender: Gender,
    val status: Status,
    val cost: Int,
    val imageUrl: String?,
    val sports: Sports,
    val town: Town,
    val createdAt: LocalDateTime?,
    val updatedAt: LocalDateTime?
) {
    companion object {
        fun from(post: Post): UpdatePostResponseDto =
            UpdatePostResponseDto(
                post.title,
                post.description,
                post.location,
                post.date,
                post.maxPeople,
                post.gender,
                post.status,
                post.cost,
                post.imageUrl,
                post.sports,
                post.town,
                post.createdAt,
                post.updatedAt
            )
    }
}
